src/Tribe/Integrations/WPML/Meta.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * Translate post ids in Event meta data.
+ *
+ * @since TBD
+ */
+class Tribe__Events__Integrations__WPML__Meta {
+
+	/**
+	 * @since TBD
+	 *
+	 */
+	public function hook() {}
+
+	/**
+	 * Translates post id in the Event meta data.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $value
+	 * @param int    $object_id
+	 * @param string $meta_key
+	 *
+	 * @return string The translated id for _EventOrganizerID & _EventVenueID.
+	 */
+	public function translate_post_id( $value, $object_id, $meta_key ) {
+
+		if ( isset( $_POST ) ) {
+			return $value;
+		}
+
+		$accepted_values = array( '_EventOrganizerID', '_EventOrganizerID_Order', '_EventVenueID' );
+
+		if ( ! in_array( $meta_key, $accepted_values ) ) {
+			return $value;
+		}
+
+		$value = $this->get_post_meta( $object_id, $meta_key );
+
+		if ( empty( $value ) ) {
+			return false;
+		}
+
+		$type = strpos( 'Organizer', $meta_key )
+			? Tribe__Events__Organizer::POSTTYPE
+			: Tribe__Events__Venue::POSTTYPE;
+
+		foreach ( $value as & $post_id ) {
+			if ( is_serialized( $post_id ) ) {
+				$array = (array) unserialize( $post_id );
+				foreach ( $array as & $id ) {
+					/**
+					 * Returns an element’s ID in the current language or in another specified language.
+					 * @param int     $id    The ID of the post type or taxonomy term to filter
+					 * @param string  $type  The type of element the ID belongs to.
+					 * @param bool    true   If set to true it will always return a value (the original value, if translation is missing)
+					 */
+					$id = apply_filters( 'wpml_object_id', $id, $type, true );
+				}
+				$post_id = $array;
+			} else {
+				/**
+				 * Returns an element’s ID in the current language or in another specified language.
+				 * @param int     $post_id   The ID of the post type or taxonomy term to filter
+				 * @param string  $type      The type of element the ID belongs to.
+				 * @param bool    true       If set to true it will always return a value (the original value, if translation is missing)
+				 */
+				$post_id = apply_filters( 'wpml_object_id', $post_id, $type, true );
+			}
+		}
+
+		return $value;
+	}
+
+	/**
+	 * Get meta value skipping filters (using direct DB query).
+	 *
+	 * @since TBD
+	 *
+	 * @param type $post_id
+	 * @param type $meta_key
+	 * @return type
+	 */
+	private function get_post_meta( $post_id, $meta_key ) {
+		global $wpdb;
+
+		return $wpdb->get_col( $wpdb->prepare(
```

Comment:
Curios here on why `get_post_meta` was not used instead?

Reply:
Due to usage of filtering on `get_post_meta` on `pre_get_metadata`. Which would trigger an infinite loop.

Reply:
Also curious, not my code, I'm just re-submitting the changes we got from wmpl in order to make it pass QA. The idea there, from what I can perceive, is that they want to get the post meta directly from DB avoiding filters, because perhaps they're already conscious they have other filters that can mess things up (perhaps they have a similar filter hooked on their core or something like that). It actually doesn't trigger an infinite loop.